# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1317315962 25200
#      Thu Sep 29 10:06:02 2011 -0700
# Branch stable
# Node ID 248d05c413dc47be312c29ad2a06896a3617639c
# Parent  00c5d8d3ee004b71a39e70d291f6ff817d2388bc
Validate inputs to line() to prevent corrupting graphic toolkit (Bug #32345)

* __line__.m: Validate number of X, Y, and Z points are equal before passing
to plotting toolkit.

diff --git a/scripts/plot/private/__line__.m b/scripts/plot/private/__line__.m
--- a/scripts/plot/private/__line__.m
+++ b/scripts/plot/private/__line__.m
@@ -40,18 +40,22 @@ function h = __line__ (p, varargin)
       num_data_args = 3;
     else
       num_data_args = 2;
     endif
   else
     num_data_args = 0;
   endif
 
+  if (num_data_args > 0 && ! size_equal (varargin{1:num_data_args}))
+    error ("line: number of X, Y, and Z points must be equal");
+  endif
+
   if (rem (nvargs - num_data_args, 2) != 0)
-    print_usage ("line");
+    error ("line: invalid number of PROPERTY / VALUE pairs");
   endif
 
   data_args = {};
   if (num_data_args > 1)
     data_args(1:4) = { "xdata", varargin{1}, "ydata", varargin{2} };
     if (num_data_args == 3)
       data_args(5:6) = { "zdata", varargin{3} };
     endif
